src/sortedvector.jl

```diff
@@ -0,0 +1,78 @@
+"""
+    SortedVector
+
+A wrapper for sorted vectors, designed for fast unions.
+
+# Constructor
+
+    SortedVector(data::AbstractVector; already_sorted=false)
+
+# Example
+
+```jldoctest
+x = SortedVector([3, 4, 2])
+x = SortedVector([1, 3, 5]; already_sorted=true)
+z = union(x, y)
+
+# output
+
+SortedVector([1, 2, 3, 4, 5])
+````
+"""
+struct SortedVector{T,V<:AbstractVector{T}} <: AbstractVector{T}
+    data::V
+
+    function SortedVector{T,V}(data::V; already_sorted=false) where {T,V<:AbstractVector{T}}
+        if already_sorted
+            new{T,V}(data)
+        else
+            new{T,V}(sort(data))
+        end
+    end
+end
+
+Base.eltype(::SortedVector{T}) where {T} = T
+Base.size(v::SortedVector) = size(v.data)
+Base.getindex(v::SortedVector, i) = v.data[i]
+Base.IndexStyle(::Type{SortedVector{T,V}}) where {T,V} = IndexStyle(V)
+
+function SortedVector(data::V; already_sorted=false) where {T,V<:AbstractVector{T}}
+    return SortedVector{T,V}(data; already_sorted)
+end
+
+Base.show(io::IO, sv::SortedVector) = print(io, "SortedVector($(sv.data))")
+
+function Base.union(v1::SortedVector{T,V}, v2::SortedVector{T,V}) where {T,V}
+    left, right = v1.data, v2.data
+    both = similar(left, length(left) + length(right))
```

Comment:
I like `both` because it sits in the middle between `left` and `right`. And `union` is absurd, we're literally inside the `union` function

This is performance-independent bikeshedding, we can do that after NeurIPS

Reply:
The name `both` confused me while reviewing and I have to maintain this code.

Reply:
if it's confusing and not just esthetic that's different

replaced it with `result`